### ISSUE TITLE
Getting full_text from truncated tweets

### ISSUE THREAD

**Issue by jmlipman** [bear/python-twitter#491]
Hi guys,

I realized that when using this library I get truncated tweets and I tried to find a solution. I found on the Internet that there is a parameter called "tweet_mode" that you are supposed to tunned as "extended". However, there is no such parameter in the GetSearch function.

I realized that there is a parameter called raw_query, and I also tried to write something like "q=whatever&tweet_mode=extended" but it didn't work.

I made a workaround and I want to share it because someone might find it useful. You just need to modify a couple of lines:

In the file models.py
In the function NewFromJsonDict from Status
Line 469 (below user_mentions=None)
add: **full_text = None** 

Line 478 (below current_user_retweet = data['current_user_retweet']['id'])
add:         **if 'full_text' in data:
            full_text = data['full_text']**

Line (last):
add full_text in the parameters such that:
return super(cls, cls).NewFromJsonDict(data=data,
                                               current_user_retweet=current_user_retweet,
                                               hashtags=hashtags,
                                               media=media,
                                               retweeted_status=retweeted_status,
                                               urls=urls,
                                               user=user,
                                               user_mentions=user_mentions,**full_text=full_text**)

This is just a workaround but it would be great if it's implemented in the library.

Regards,
JM.

**Comment by mejmo** [comment]
This does not work for me either.. When truncated True is, I did not have any property full_text in data..

**Comment by zefoo** [comment]
I'm seeing this too. This may be helpful: https://github.com/bear/python-twitter/issues/506 - Although, I can't seem to get the workaround to work. Maybe you'll have some better luck? I'm curious if you find the solution.

**Comment by negi111111** [comment]
Im also looking for it. Did you find a good solution?

**Comment by techieshark** [comment]
One thing: if anyone else runs into this, make sure you have `tweet_mode='extended'` when you call `twitter.Api()`. I got thrown off for a while by that.

```
twitter.Api(
            consumer_key= os.environ['CONSUMER_KEY'],
            consumer_secret= os.environ['CONSUMER_SECRET'],
            access_token_key= os.environ['ACCESS_TOKEN'],
            access_token_secret= os.environ['ACCESS_TOKEN_SECRET'],
            cache=None,
            tweet_mode= 'extended')
```

**Comment by jeremylow** [comment]
Fix lands in 564e6bb620b20d09d937917e3094bcdb2c82c812
